### PR TITLE
05432 use pbj in hedera network service

### PR DESCRIPTION
**Pull request by povolev15**
integrate PBJ ReadableSequentialData WritableSequentialData and fix all issues with unit tests

Also ran spotless on top of the latest merge

#5622

### CODE REVIEW

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoContextAdapterCodec.java [hunk 1]
```diff
@@ -23,28 +23,31 @@
 import com.swirlds.common.io.streams.SerializableDataInputStream;
 import com.swirlds.common.io.streams.SerializableDataOutputStream;
 import edu.umd.cs.findbugs.annotations.NonNull;
-import java.io.IOException;
+import java.io.ByteArrayInputStream;import java.io.ByteArrayOutputStream;import java.io.IOException;
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoContextAdapterCodec.java:26]: three lines collapsed to one...

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoRunningHashesAdapterCodec.java [hunk 2]
```diff
@@ -23,28 +23,31 @@
 import com.swirlds.common.io.streams.SerializableDataInputStream;
 import com.swirlds.common.io.streams.SerializableDataOutputStream;
 import edu.umd.cs.findbugs.annotations.NonNull;
-import java.io.IOException;
+import java.io.ByteArrayInputStream;import java.io.ByteArrayOutputStream;import java.io.IOException;
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoRunningHashesAdapterCodec.java:26]: collapsed lines

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoSpecialFilesAdapterCodec.java [hunk 3]
```diff
@@ -23,28 +23,34 @@
...
-            item.serialize(out);
-        } else {
-            throw new IllegalArgumentException("Expected a SerializableDataOutputStream");
-        }
+    public void write(final @NonNull MerkleSpecialFiles item, final @NonNull WritableSequentialData output)
+            throws IOException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        SerializableDataOutputStream sdo = new SerializableDataOutputStream(baos);
+        item.serialize(sdo);
+        sdo.flush();
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoSpecialFilesAdapterCodec.java:50]: This calls baos.flush, so the next line is unnecessary.

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java [hunk 4]
```diff
@@ -20,12 +20,9 @@
 import com.hedera.node.app.service.mono.state.submerkle.ExchangeRates;
 import com.hedera.node.app.service.mono.state.submerkle.SequenceNumber;
 import com.hedera.node.app.service.network.impl.serdes.MonoContextAdapterCodec;
-import com.hedera.pbj.runtime.io.ReadableSequentialData;
-import com.hedera.pbj.runtime.io.WritableSequentialData;
-import com.swirlds.common.io.streams.SerializableDataInputStream;
+import com.hedera.pbj.runtime.io.stream.ReadableStreamingData;import com.hedera.pbj.runtime.io.stream.WritableStreamingData;import com.swirlds.common.io.streams.SerializableDataInputStream;
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java:23]: another collapsed set of imports.

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java [hunk 5]
```diff
@@ -40,38 +37,28 @@ class MonoContextAdapterSerdesTest {
...
-    private WritableSequentialData output;
 
     final MonoContextAdapterCodec subject = new MonoContextAdapterCodec();
 
     @Test
     void doesntSupportUnnecessary() {
         assertThrows(UnsupportedOperationException.class, () -> subject.measureRecord(SOME_CONTEXT));
-        assertThrows(UnsupportedOperationException.class, () -> subject.measure(input));
-        assertThrows(UnsupportedOperationException.class, () -> subject.fastEquals(SOME_CONTEXT, input));
+        assertThrows(UnsupportedOperationException.class, () -> subject.measure(null));
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java:46]: This changes the effect of the test, which may not have been intentional...

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoRunningHashesAdapterSerdesTest.java [hunk 6]
```diff
@@ -16,60 +16,47 @@
...
-    private WritableSequentialData output;
-
     final MonoRunningHashesAdapterCodec subject = new MonoRunningHashesAdapterCodec();
 
     @Test
     void doesntSupportUnnecessary() {
         assertThrows(UnsupportedOperationException.class, () -> subject.measureRecord(SOME_HASHES));
-        assertThrows(UnsupportedOperationException.class, () -> subject.measure(input));
-        assertThrows(UnsupportedOperationException.class, () -> subject.fastEquals(SOME_HASHES, input));
+        assertThrows(UnsupportedOperationException.class, () -> subject.measure(null));
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoRunningHashesAdapterSerdesTest.java:46]: This also changes the effect.  If the intent is to ensure these methods always throw, then perhaps calling with both null and non-null inputs would be advisable.

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoSpecialFilesAdapterSerdesTest.java [hunk 7]
```diff
@@ -45,19 +45,13 @@ class MonoSpecialFilesAdapterSerdesTest {
...
-    private WritableSequentialData output;
-
     final MonoSpecialFilesAdapterCodec subject = new MonoSpecialFilesAdapterCodec();
 
     @Test
     void doesntSupportUnnecessary() {
         assertThrows(UnsupportedOperationException.class, () -> subject.measureRecord(SOME_SPECIAL_FILES));
-        assertThrows(UnsupportedOperationException.class, () -> subject.measure(input));
-        assertThrows(UnsupportedOperationException.class, () -> subject.fastEquals(SOME_SPECIAL_FILES, input));
+        assertThrows(UnsupportedOperationException.class, () -> subject.measure(null));
```
Comment (jsync-swirlds) [inline hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoSpecialFilesAdapterSerdesTest.java:53]: This also changes the effect. If the intent is to ensure these methods always throw, then perhaps calling with both null and non-null inputs would be advisable.

**Comment by jsync-swirlds** [comment]
Lots of format changes in unrelated code, worth taking note of that as it's the majority of the files affected and may cause more merge conflicts.

**Comment by povolev15** [comment]
> Lots of format changes in unrelated code, worth taking note of that as it's the majority of the files affected and may cause more merge conflicts.

You are right my main changes in services module. But I run spotless on top of my code and the merged code

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java (hunk 4 above)
Comment (mhess-swl) [inline hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java:23]: formatting error here

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoContextAdapterCodec.java (hunk 1 above)
Comment (mhess-swl) [inline hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoContextAdapterCodec.java:26]: Formatting error here

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoRunningHashesAdapterCodec.java (hunk 2 above)
Comment (mhess-swl) [inline hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoRunningHashesAdapterCodec.java:26]: More formatting here

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java [hunk 8]
```diff
@@ -117,15 +118,14 @@ public void handle(
...
+        final var maybeTopic = requireNonNull(topicStore)
+                .get(topicUpdate.topicIDOrElse(TopicID.DEFAULT).topicNum());
         validateTrue(maybeTopic.isPresent(), INVALID_TOPIC_ID);
         final var topic = maybeTopic.get();
         validateFalse(topic.deleted(), INVALID_TOPIC_ID);
 
         // First validate this topic is mutable; and the pending mutations are allowed
-        validateFalse(
-                topic.adminKey() == null && wantsToMutateNonExpiryField(topicUpdate), UNAUTHORIZED);
+        validateFalse(topic.adminKey() == null && wantsToMutateNonExpiryField(topicUpdate), UNAUTHORIZED);
```
[the remark this answers is not in the document]
Comment (povolev15) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:128]: interesting I missed it  I will check

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java [hunk 9]
```diff
@@ -309,10 +313,7 @@ private boolean isNotNeeded(@NonNull final ContractID id) {
      * @return true if the given accountID is
      */
     private boolean designatesAccountRemoval(@NonNull final AccountID id) {
-        return id.shardNum() == 0
-                && id.realmNum() == 0
-                && id.accountNumOrElse(0L) == 0
-                && !id.hasAlias();
+        return id.shardNum() == 0 && id.realmNum() == 0 && id.accountNumOrElse(0L) == 0 && !id.hasAlias();
```
[the remark this answers is not in the document]
Comment (povolev15) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:316]: probably base on the spotless configuration

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (povolev15) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:128]: maybe worth to review the spotless how it works

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java (hunk 5 above)
Comment (povolev15) [reply hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoContextAdapterSerdesTest.java:46]: I fix it thank for the comment

File: hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoRunningHashesAdapterSerdesTest.java (hunk 6 above)
Comment (povolev15) [reply hedera-node/hedera-network-service-impl/src/test/java/com/hedera/node/app/service/network/impl/test/serdes/MonoRunningHashesAdapterSerdesTest.java:46]: I fixed it thank you

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoSpecialFilesAdapterCodec.java (hunk 3 above)
Comment (povolev15) [reply hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoSpecialFilesAdapterCodec.java:50]: ok fixed

File: hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoContextAdapterCodec.java (hunk 1 above)
Comment (povolev15) [reply hedera-node/hedera-network-service-impl/src/main/java/com/hedera/node/app/service/network/impl/serdes/MonoContextAdapterCodec.java:26]: thx

File: hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java (hunk 9 above)
[the remark this answers is not in the document]
Comment (povolev15) [reply hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/workflows/PreHandleContext.java:316]: I fixed it and rerun gradle and it is agin return it to be as it is

File: hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java (hunk 8 above)
[the remark this answers is not in the document]
Comment (povolev15) [reply hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java:128]: same story

**Review by iwsimon — APPROVED** [review]
LGTM

**Review by jsync-swirlds — APPROVED** [review]

**Pull request merged**